### PR TITLE
Update valid HTTP methods in the type definitions

### PR DESCRIPTION
**Pull request by NicholasBoll**
The implementation of cy.request was recently changed (#3555) to support more HTTP methods, but the type definitions were not updated to match.

Adds type definitions for closed issue #1302

### CODE REVIEW

File: cli/types/index.d.ts [hunk 1]
```diff
@@ -29,7 +29,7 @@ declare const assert: Chai.AssertStatic
 declare namespace Cypress {
   type FileContents = string | any[] | object
   type HistoryDirection = "back" | "forward"
-  type HttpMethod = "GET" | "POST" | "PUT" | "DELETE" | "OPTIONS" | "HEAD" | "TRACE" | "CONNECT" | "PATCH"
+  type HttpMethod = "GET" | "POST" | "PUT" | "DELETE" | "PATCH" | "OPTIONS" | "HEAD" | "TRACE" | "COPY" | "LOCK" | "MKCOL" | "MOVE" | "PURGE" | "PROPFIND" | "PROPPATCH" | "UNLOCK" | "REPORT" | "MKACTIVITY" | "CHECKOUT" | "MERGE" | "M-SEARCH" | "NOTIFY" | "SUBSCRIBE" | "UNSUBSCRIBE" | "SEARCH" | "CONNECT"
```
Comment (NicholasBoll) [inline cli/types/index.d.ts]: I wonder if this should just be `string`. From what I can tell, node's type definition just use `string`: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/v8/index.d.ts#L1032

**Comment by jennifer-shehane** [comment]
@NicholasBoll Perhaps - see the implementation we are using to check methods https://github.com/jshttp/methods/blob/master/index.js

**Comment by NicholasBoll** [comment]
That runtime implementation calls `toLowerCase` and checks the method. The type definitions now require only uppercase. If the methods are enumerated, we may want both uppercase and lowercase to be represented. It is unlikely a user would mix casing (ex: `PaTcH`). What do you think?

File: cli/types/index.d.ts (hunk 1 above)
Comment (flotwig) [reply cli/types/index.d.ts]: Technically speaking, this list of methods could change with any new Node.js release. We also accept mixed casing like you've noted, so technically this list of valid method names would be a looooooot longer if it encompassed everything.

I think `string` would be fine here.

**Pull request merged**